### PR TITLE
Add mw.Title types

### PR DESCRIPTION
**Pull request by jdlrobson**

### CODE REVIEW

File: MediaWiki.d.ts [hunk 1]
```diff
@@ -2,6 +2,12 @@ interface MwApi {
 	saveOption( name: string, value: unknown ): JQuery.Promise<any>;
 }
 
+interface MwTitle {
+	getUrl(): string
```
Comment (DannyS712) [inline MediaWiki.d.ts:6]: getUrl takes an optional object parameter

We should also be consistent one way or the other about the capitalization of "number" or "Number" - its fine to add whichever one you want in this patch, but afterwards we should standardize

File: MediaWiki.d.ts [hunk 2]
```diff
@@ -164,7 +170,9 @@ interface MediaWiki {
 	 */
 	now(): number,
 
-	requestIdleCallback( callback: Function ): () => void
+	requestIdleCallback( callback: Function ): () => void,
```
Comment (DannyS712) [inline MediaWiki.d.ts:173]: Should be included in commit message

**Review by DannyS712 — CHANGES_REQUESTED** [review]

File: MediaWiki.d.ts (hunk 1 above)
Comment (jdlrobson) [reply MediaWiki.d.ts]: Done. Noted about number vs Number. I think "number" is definitely the right choice considering over repos.

File: MediaWiki.d.ts (hunk 2 above)
Comment (jdlrobson) [reply MediaWiki.d.ts]: Done.

Comment (DannyS712) [reply MediaWiki.d.ts]: The overall title for this is still "Add mw.Title types"

File: MediaWiki.d.ts [hunk 3]
```diff
@@ -170,9 +170,8 @@ interface MediaWiki {
 	 */
 	now(): number,
 
-	requestIdleCallback( callback: Function ): () => void,
+	requestIdleCallback( callback: Function ): () => void
```
Comment (DannyS712) [inline MediaWiki.d.ts:173]: shouldn't this still have a comma afterwards?

File: MediaWiki.d.ts (hunk 2 above)
Comment (jdlrobson) [reply MediaWiki.d.ts]: I'm confused. I'm not touching this line any more. requestIdleCallback has nothing to do with mw.Title types.

File: MediaWiki.d.ts (hunk 3 above)
Comment (jdlrobson) [reply MediaWiki.d.ts:173]: I'm a bit confused about your early change. I thought you were asking for me to move the definition.

Note this line is not being touched as part of this change as it's unrelated.

File: MediaWiki.d.ts (hunk 2 above)
Comment (DannyS712) [reply MediaWiki.d.ts]: Oh - I guess I'm confused by the github interface, I thought it was still be included in the overall change

File: MediaWiki.d.ts (hunk 3 above)
Comment (DannyS712) [reply MediaWiki.d.ts:173]: Sorry, ignore me, I got confused by the github interface

**Pull request merged**